File: 1-Two-Sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        hashMap = {}
        for i in range(len(nums)):
            if target - nums[i] in hashMap:
                return [i, hashMap[(target - nums[i])]]
            hashMap[nums[i]] = i

File: 1-Two-Sum.ts
function twoSum(nums: number[], sum: number):number[] {
    const hashMap = new Map<number, number>();
    for (let i = 0; i < nums.length; i++) {
        const diff = sum - nums[i];
        if (hashMap.has(diff)) {
            return [hashMap.get(diff), i];
        }
        hashMap.set(nums[i], i);
    }
}